### DISCUSSION TITLE
It's spend a lot time loading data

### DISCUSSION THREAD

**Discussion by minhldhe130866** [TanStack/table#3563] [Q&A]
My project used react-table including 600 records, this number is not too big. And I load fully batch of 600 records. My manament site takes nearly 1 minute to load and show table. What's my problem ? And how to solve it. Thanks in advance

**Comment by Kepro** [comment]
600 records? and how many columns? any custom cells? are you using memo for columns and data?

**Reply by minhldhe130866** [reply]
Thank you for helping, I found the good way already. Last time, I just wanted to add inline button like edit, view, or delete. I fetched API from server, and used map() to match each ID of record with ID of button in order to get ID of record after I clicked button of that record. Now I use  keyword "Cell : " ( Idn what to say clearly, sr ) when I define colums of data before passing to table. By the way, could you please explain to me that why we must use useMemo to react table. Thanks in advance  !!!

**Reply by Kepro** [reply]
if our `data` object will change on every render it will kill the performance, so we want to keep data structure memoized as much as we can ( https://reactjs.org/docs/hooks-reference.html#usememo )
